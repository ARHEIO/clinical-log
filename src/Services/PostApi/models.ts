export interface Post {
  name: Name; // name of user posting
  time: string; // date of posting
  content: string;
}

export interface Name {
  firstName: string;
  lastName: string;
}

export interface Reacts {
  like: number;
  haha: number;
  wow: number;
  sad: number;
}

export interface PublicPost extends Post {
  isVenting: boolean;
  reacts: Reacts;
}
