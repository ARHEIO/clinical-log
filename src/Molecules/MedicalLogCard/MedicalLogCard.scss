.medical-log-top {
  display: flex;
  flex-direction: row;
  justify-content: space-between;

  &__name {
    align-items: center;
    display: flex;
    flex-direction: row;
    justify-content: center;

    & > p {
      margin-left: 16px;
    }
  }

  &__date {
    align-items: flex-end;
    display: flex;
    flex-direction: column;
    justify-content: center;

    *:first-child {
      padding-bottom: 8px;
    }
  }
}

.medical-log {
  &__content {
    padding: 16px 0;
  }
}
