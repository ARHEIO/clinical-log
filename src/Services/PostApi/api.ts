/* eslint-disable max-len */
/* eslint-disable object-curly-newline */
import { PublicPost, Post } from './models';

const publicPosts: PublicPost[] = [
  {
    name: {
      firstName: 'Enzo',
      lastName: 'Brooks',
    },
    time: new Date().toString(),
    content: 'Up to one-third of patients experience markers of distress during palliative ventilator withdrawal, study finds',
    isVenting: false,
    reacts: { wow: 12, haha: 0, like: 5, sad: 0 },
  },
  {
    name: {
      firstName: 'Chris',
      lastName: 'Redfield',
    },
    time: '2020-06-15T03:40:44.816Z',
    content: 'Scientists have discovered a population of neurons that control hibernation-like behavior, or torpor, in mice',
    isVenting: false,
    reacts: { wow: 0, haha: 0, like: 12, sad: 0 },
  },
  {
    name: {
      firstName: 'Chris',
      lastName: 'Redfield',
    },
    time: '2020-06-14T03:40:44.816Z',
    content: 'Scientists have discovered a population of neurons that control hibernation-like behavior, or torpor, in mice',
    isVenting: false,
    reacts: { wow: 0, haha: 0, like: 12, sad: 0 },
  },
  {
    name: {
      firstName: 'Enzo',
      lastName: 'Brooks',
    },
    time: '2020-06-12T01:30:35.096Z',
    content: 'Today, @Jerry is officially cancer free!',
    isVenting: true,
    reacts: { wow: 0, haha: 0, like: 12, sad: 0 },
  },
];
const clinicalLogs: Post[] = [
  {
    name: {
      firstName: 'Chris',
      lastName: 'Redfield',
    },
    time: '2020-06-12T01:30:35.096Z',
    content: 'Patient in room 78 blood pressure is 128/70',
  },
  {
    name: {
      firstName: 'Samantha',
      lastName: 'Yule',
    },
    time: '2020-06-12T01:30:35.096Z',
    content: 'Patient in room 47 is showing decreased signs of self awareness',
  },
  {
    name: {
      firstName: 'Samantha',
      lastName: 'Yule',
    },
    time: '2020-06-12T01:30:35.096Z',
    content: 'Patient in room 78 blood pressure is 134/96',
  },
  {
    name: {
      firstName: 'Enzo',
      lastName: 'Brooks',
    },
    time: '2020-06-12T01:30:35.096Z',
    content: 'Patient in room 78 blood pressure is 139/104',
  },
  {
    name: {
      firstName: 'Samantha',
      lastName: 'Yule',
    },
    time: '2020-06-12T01:30:35.096Z',
    content: 'Patient in room 78 blood pressure is 119/79',
  },
];

const successfulCall = (): boolean => (Math.round(Math.random() * 100)) < 95; // simulating a network failure

export const getPublicPosts = (): Promise<PublicPost[]> => new Promise((resolve, reject) => setTimeout(() => (successfulCall() ? resolve(publicPosts) : reject()), 1000));

export const getClinicalLog = (): Promise<Post[]> => new Promise((resolve, reject) => setTimeout(() => (successfulCall() ? resolve(clinicalLogs) : reject()), 1000));
