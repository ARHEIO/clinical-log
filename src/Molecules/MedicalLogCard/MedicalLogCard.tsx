import React, { ReactElement } from 'react';
import './MedicalLogCard.scss';
import { Post } from '../../Services/PostApi/models';
import Card from '../../Atoms/Card/Card';
import IconContainer from '../../Atoms/IconContainer/IconContainer';
import TimeContainer from '../../Atoms/TimeContainer/TimeContainer';

// Todo, wrap @mentions in tags to make them show up as blue

const MedicalLogCard = ({ post }: {post: Post}): ReactElement => (
  <Card className="public-post">
    <div className="public-post-top">
      <div className="public-post-top__name">
        <IconContainer username={post.name} />
        <p>{`${post.name.firstName} ${post.name.lastName}`}</p>
      </div>
      <div className="public-post-top__date">
        <TimeContainer time={post.time} relative={false} />
      </div>
    </div>
    <p className="public-post__content">{post.content}</p>
  </Card>
);

export default MedicalLogCard;
