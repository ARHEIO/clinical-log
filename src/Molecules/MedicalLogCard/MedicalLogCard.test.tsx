import React from 'react';
import { render } from '@testing-library/react';
import pretty from 'pretty';
import MedicalLogCard from './MedicalLogCard';
import { Post } from '../../Services/PostApi/models';

const defaultPost: Post = {
  name: {
    firstName: 'Chris',
    lastName: 'Redfield',
  },
  time: new Date().toISOString(),
  content: 'Scientists have discovered a population of neurons that control hibernation-like behavior, or torpor, in mice',
};

it('renders learn react link', () => {
  const { container } = render(<MedicalLogCard post={defaultPost} />);
  expect(pretty(container.innerHTML)).toMatchSnapshot();
});
