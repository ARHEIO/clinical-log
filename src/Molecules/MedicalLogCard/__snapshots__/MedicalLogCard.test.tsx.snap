// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
"<div class=\\"card-container medical-log\\">
  <div class=\\"medical-log-top\\">
    <div class=\\"medical-log-top__name\\">
      <div class=\\"icon-container\\">
        <p>CR</p>
      </div>
      <p>Chris Redfield</p>
    </div>
    <div class=\\"medical-log-top__date\\">
      <p>TimeContainer</p>
    </div>
  </div>
  <p class=\\"medical-log__content\\">Scientists have discovered a population of neurons that control hibernation-like behavior, or torpor, in mice</p>
</div>"
`;
